test/Atmos/Model/ref_state.jl

```diff
@@ -0,0 +1,82 @@
+using CLIMAParameters
+using CLIMAParameters.Planet: R_d, grav, MSLP
+using StaticArrays
+
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+using ClimateMachine
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.Mesh.Geometry
+using ClimateMachine.Thermodynamics
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.Atmos
+using ClimateMachine.ConfigTypes
+using ClimateMachine.DGMethods
+using ClimateMachine.DGMethods: vars_state_auxiliary, number_state_auxiliary
+using ClimateMachine.DGMethods: BalanceLaw, LocalGeometry
+using ClimateMachine.MPIStateArrays
+using ClimateMachine.VariableTemplates
+
+using Test
+using ForwardDiff
+
+# We should provide an interface to call all physics
+# kernels in some way similar to this:
+function compute_ref_state(z::FT, atmos) where {FT}
```

Comment:
@kpamnany @simonbyrne we could make spatially-continuous unit tests if we somehow provide wrappers like this around our unit tests (earlier I used this with `ForwardDiff` to test hydrostatic balance, but that was moved to `TemperatureProfiles`).